crates/api/src/func.rs

```diff
@@ -970,7 +971,7 @@ pub trait IntoFunc<Params, Results> {
 /// the caller's memory until interface types has been fully standardized and
 /// implemented.
 pub struct Caller<'a> {
-    store: &'a Store,
+    store: &'a Weak<StoreInner>,
```

Comment:
Could you add a brief comment here mentioning that the `Weak` is just to break a reference cycle, and that it's always meant to be upgradable in practice?